src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java

```diff
@@ -486,6 +486,16 @@ private void cleanupOldData(SnapshotConfig config, TransactionPruner transaction
             throw new SnapshotException("could not add the cleanup job to the transaction pruner", e);
         }
     }
+    
+    private int getStartingIndex(SnapshotConfig config) {
+        MilestoneViewModel milestonevm = null;
+        try {
+            milestonevm = MilestoneViewModel.first(tangle);
+        } catch (Exception e) {
+            log.error("Can't load the first milestone from the db", e);
+        }
+        return milestonevm != null ? milestonevm.index() : config.getMilestoneStartIndex() + 1;
```

Comment:
I am trying to think whether this is the correct behavior...
If we fail to read from db perhaps we should panic and throw the exception?

What's your opinion?

Anyhows, since this inline with previous behavior and no change should be part of the PR then please either:
1. Fix in a separate PR
2. Open an issue if you don't want to do it now
3. Write a comment why the current behavior is good

Reply:
ive changed it so that it throws a snapshotexception! so no snapshot will be taken at least